core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -465,6 +479,27 @@ public Table loadTable(SessionContext context, TableIdentifier identifier) {
     return table;
   }
 
+  private RestTable tableSupportsRemoteScanPlanning(
+      TableOperations ops, TableIdentifier finalIdentifier, RESTClient restClient) {
+    if (ops.current().properties().containsKey(REST_TABLE_SCAN_PLANNING_PROPERTY)) {
+      boolean tableSupportsRemotePlanning =
+          ops.current().propertyAsBoolean(REST_TABLE_SCAN_PLANNING_PROPERTY, false);
+      if (tableSupportsRemotePlanning && restServerPlanningEnabled) {
+        return new RestTable(
+            ops,
+            fullTableName(finalIdentifier),
+            metricsReporter(paths.metrics(finalIdentifier), restClient),
+            this.client,
+            catalogAuth,
```

Comment:
might need to pass the auth session based on context passed in the loadTable

Reply:
Yeah, more specifically I think this should be the `tableSession` whose parent session is the `contextualSession`